src/main.cpp

```diff
@@ -11,124 +11,126 @@
-  constexpr auto width = 800;
-  constexpr auto height = 480;
+  constexpr auto width = buildparams::output_width;
+  constexpr auto height = buildparams::output_height;
 
   // Allocate frame buffer on host
-  std::vector<color> fb(width * height);
+  std::array<color, (width * height)> fb;
 
   /// Graphical objects
   std::vector<hittable_t> hittables;
 
   // Generating a checkered ground and some random spheres
   texture_t t =
-      checker_texture(color { 0.2f, 0.3f, 0.1f }, color { 0.9f, 0.9f, 0.9f });
+	  checker_texture(color { 0.2f, 0.3f, 0.1f }, color { 0.9f, 0.9f, 0.9f });
   material_t m = lambertian_material(t);
   hittables.emplace_back(sphere(point { 0, -1000, 0 }, 1000, m));
   t = checker_texture(color { 0.9f, 0.9f, 0.9f }, color { 0.4f, 0.2f, 0.1f });
 
+  LocalPseudoRNG rng;
+
   for (int a = -11; a < 11; a++) {
-    for (int b = -11; b < 11; b++) {
-      // Based on a random variable , the material type is chosen
-      auto choose_mat = random_float();
-      // Spheres are placed at a point randomly displaced from a,b
-      point center(a + 0.9f * random_float(), 0.2f, b + 0.9f * random_float());
-      if (sycl::length((center - point(4, 0.2f, 0))) > 0.9f) {
-        if (choose_mat < 0.4f) {
-          // Lambertian
-          auto albedo = randomvec() * randomvec();
-          hittables.emplace_back(
-              sphere(center, 0.2f, lambertian_material(albedo)));
-        } else if (choose_mat < 0.8f) {
-          // Lambertian movig spheres
-          auto albedo = randomvec() * randomvec();
-          auto center2 = center + point { 0, random_float(0, 0.25f), 0 };
-          hittables.emplace_back(sphere(center, center2, 0.0f, 1.0f, 0.2f,
-                                        lambertian_material(albedo)));
-        } else if (choose_mat < 0.95f) {
-          // metal
-          auto albedo = randomvec(0.5f, 1);
-          auto fuzz = random_float(0, 0.5f);
-          hittables.emplace_back(
-              sphere(center, 0.2f, metal_material(albedo, fuzz)));
-        } else {
-          // glass
-          hittables.emplace_back(
-              sphere(center, 0.2f,
-                     dielectric_material(1.5f, color { 1.0f, 1.0f, 1.0f })));
-        }
-      }
-    }
+	for (int b = -11; b < 11; b++) {
+	  // Based on a random variable , the material type is chosen
+	  auto choose_mat = random_float(rng);
+	  // Spheres are placed at a point randomly displaced from a,b
+	  point center(a + 0.9f * random_float(rng), 0.2f, b + 0.9f * random_float(rng));
+	  if (sycl::length((center - point(4, 0.2f, 0))) > 0.9f) {
+		if (choose_mat < 0.4f) {
+		  // Lambertian
+		  auto albedo = randomvec(rng) * randomvec(rng);
+		  hittables.emplace_back(
+			  sphere(center, 0.2f, lambertian_material(albedo)));
+		} else if (choose_mat < 0.8f) {
+		  // Lambertian movig spheres
+		  auto albedo = randomvec(rng) * randomvec(rng);
+		  auto center2 = center + point { 0, random_float(0, 0.25f, rng), 0 };
```

Comment:
I am thinking to a terser `rng.vec()` and `rng.float_t()` syntax.

Reply:
Good idea. Implemented in 2216dfd

Reply:
438cd08  after rebase